### PR TITLE
Add preset sync controller

### LINKED ISSUE #9465: Create preset sync controller
Create a sync controller to synchronize prests between master nad seeds.
Based on the preset changes another preset controller located on the seed will update the user cluster annotations to indicate preset changes. The UI will notify users about the preset which was used for cluster creation. It will allow making some actions with the clusters.

### PR DESCRIPTION
**Pull request by zreigz**
**What does this PR do / Why do we need it**: Add preset sync controller

**Does this PR close any issues?**:
Fixes #9465

```release-note
Add preset sync controller
```

### CODE REVIEW

**Comment by zreigz** [comment]
@xrstf PTAL

File: pkg/controller/master-controller-manager/preset-synchronizer/OWNERS [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# See the OWNERS docs: https://git.k8s.io/community/contributors/guide/owners.md
+
+approvers:
+  - sig-app-management
```
Comment (xrstf) [inline pkg/controller/master-controller-manager/preset-synchronizer/OWNERS]: I don't think sig-app should own this.

File: pkg/controller/master-controller-manager/preset-synchronizer/controller.go [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+const (
+	// This controller syncs the kubermatic preset on the master cluster to the seed clusters.
+	ControllerName = "preset_syncing_controller"
```
Comment (xrstf) [inline pkg/controller/master-controller-manager/preset-synchronizer/controller.go]: Please call it `kkp-preset-synchronizer`

**Review by xrstf — COMMENTED** [review]
Two minor comments

File: pkg/controller/master-controller-manager/preset-synchronizer/OWNERS (hunk 1 above)
Comment (zreigz) [reply pkg/controller/master-controller-manager/preset-synchronizer/OWNERS]: changed for cluster-management

File: pkg/controller/master-controller-manager/preset-synchronizer/controller.go (hunk 2 above)
Comment (zreigz) [reply pkg/controller/master-controller-manager/preset-synchronizer/controller.go]: done

**Pull request merged**